firestore/integration_test_internal/src/query_test.cc

```diff
@@ -675,24 +772,38 @@ TEST_F(QueryTest, TestQueriesCanUseInFilters) {
         {{"zip", FieldValue::Array({FieldValue::Integer(98101),
                                     FieldValue::Integer(98102)})}}}});
   // Search for zips matching 98101, 98103, or [98101, 98102].
-  QuerySnapshot snapshot = ReadDocuments(collection.WhereIn(
+  const Query query1 = collection.WhereIn(
       "zip", {FieldValue::Integer(98101), FieldValue::Integer(98103),
               FieldValue::Array(
-                  {FieldValue::Integer(98101), FieldValue::Integer(98102)})}));
+                  {FieldValue::Integer(98101), FieldValue::Integer(98102)})});
+  const QuerySnapshot snapshot = ReadDocuments(query1);
   EXPECT_EQ(std::vector<MapFieldValue>(
                 {{{"zip", FieldValue::Integer(98101)}},
                  {{"zip", FieldValue::Integer(98103)}},
                  {{"zip", FieldValue::Array({FieldValue::Integer(98101),
                                              FieldValue::Integer(98102)})}}}),
             QuerySnapshotToValues(snapshot));
 
+  const AggregateQuery aggregate_query1 = query1.Count();
+  const AggregateQuerySnapshot aggregate_snapshot1 =
+      ReadAggregate(aggregate_query1);
+  EXPECT_EQ(3, aggregate_snapshot1.count());
+  EXPECT_EQ(aggregate_query1, aggregate_snapshot1.query());
+
   // With objects.
-  snapshot = ReadDocuments(collection.WhereIn(
-      "zip", {FieldValue::Map({{"code", FieldValue::Integer(500)}})}));
+  const Query query2 = collection.WhereIn(
+      "zip", {FieldValue::Map({{"code", FieldValue::Integer(500)}})});
+  const snapshot = ReadDocuments(query2);
```

Comment:
`snapshot` needs to have a type

Reply:
Once you resolve this comment, please add the "tests requested - quick" label to this PR to trigger the full slate of tests.

Reply:
Done and triggered.